### PR TITLE
Make Pico-PIO-USB compatible with Arduino library

### PR DESCRIPTION
**Pull request by hathach**
implement #31 
- rename example to `examples` (plurals)
- add generated usb_rx/tx.pio.h header to src
- add `library.properties`: Please update library description to your liking. You could also add your maintainer email if wanted. Note: `version=` must be update correctly/consistently when making an release for arduino indexer to update
- include examples/arduino/device_info that print out device descriptor with pio-usb as host. required wip branch of 
  - https://github.com/adafruit/Adafruit_TinyUSB_Arduino/tree/rp2040-pio-usb-host
  - 120mh, 240mhz option PR https://github.com/earlephilhower/arduino-pico/pull/628 

~~Note: Since this PR also include the changes in #34, I will make this as draft and rebase when #34 is merged~~ rebased.

### CODE REVIEW

File: .github/workflows/build_arduino.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        arduino-cli core update-index --additional-urls $BSP_URL
+        arduino-cli core install rp2040:rp2040 --additional-urls $BSP_URL
+        arduino-cli lib install "Adafruit TinyUSB Library"
+        # symlink our to libraries folder
+        ln -s $GITHUB_WORKSPACE $HOME/Arduino/libraries/
+        # Library summary
+        arduino-cli lib list
+
+    #- name: Build examples
+    #  run: arduino-cli compile --warnings all --fqbn rp2040:rp2040:${{ matrix.board }}:usbstack=tinyusb examples/arduino/device_info
```
Comment (hathach) [inline .github/workflows/build_arduino.yml:52]: skip building for now, will enable ci later when we are indexed in arduino library

File: library.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+version=0.4.1
+author=sekigon-gonnoc
+maintainer=sekigon-gonnoc
+sentence=Pico PIO USB library for Arduino
+paragraph=Pico PIO USB library for Arduino
+category=Communication
+url=https://github.com/sekigon-gonnoc/Pico-PIO-USB
+architectures=rp2040
+includes=pio_usb.h
+depends=Adafruit TinyUSB Library
```
Comment (hathach) [inline library.properties:11]: let me know if you want to make any changes to the text of the library. Sentence and paragraph and author/maintainer will be displayed in the Arduino library e.g

image

Note: `version=` must be updated consistently with each new release

**Comment by hathach** [comment]
rebased and marked as ready

**Comment by hathach** [comment]
@sekigon-gonnoc thanks for merging, this repo is hopefully ready to submit to arduino indexer. Though I think you need to make an first release after the addition of library.properties. Would you mind 
- increase the `version=` in library.properties to e.g 0.5.0
- make an new release with exact version on github

**Comment by hathach** [comment]
PS: Just saw your release now, perfect !! Would you like to make an PR to register with arduino similar to this
https://github.com/arduino/library-registry/pull/907

Or I could help you, since I have a bit of exp with it.

**Comment by sekigon-gonnoc** [comment]
@hathach 
I made pull request and it was merged. Thank you for your advice.
https://github.com/arduino/library-registry/pull/1501

**Pull request merged**